### ISSUE TITLE
Section formatting for PDFs

### ISSUE THREAD

**Issue by amilesj** [benmarwick/huskydown#7]
The headers for sections and subsections are the same size and format when exporting the PDF version, making it hard to see a hierarchical structure beyond a single section. This seems to be true no matter how many "#'s" are used before each header (beyond 2). Here is a screenshot of an example: here I would like "Methods" to be a section and "Patients and Isolates" to be a subsection, but they look exactly the same. 

![screen shot 2018-05-08 at 3 19 04 pm](https://user-images.githubusercontent.com/20647069/39785969-29126342-52d3-11e8-8f11-97778f79631c.png)

Would it be possible to tap into more of LaTeX's section capabilities by using different sizes/boldness for sections and subsections, and/or allowing the option for section and subsection numbers, perhaps in the YAML header? 

Thanks!
Arianna

**Comment by benmarwick** [comment]
Thanks for your comment, that is an interesting observation that I hadn't noticed before. I agree that we should be able to have a few different levels of headings using the markdown `#`, `##`, `###` notation. 

Here's one way to get that. In your `index` directory where the Rmd files for each chapter are, there is a file `template.tex`, open that in RStudio and look for `\begin{document}`. Make some space directly below that line and paste in these lines: 
```

% Override the small caps headings so we have 
% section headers that decrease in size as we 
% use more # in the R Markdown
\makeatletter
\renewcommand\section{\@startsection {section}{1}{\z@}%
                                   {-3.5ex \@plus -1ex \@minus -.2ex}%
                                   {2.3ex \@plus.2ex}%
                                   {\normalfont\LARGE\bfseries}}% from \Large
\renewcommand\subsection{\@startsection{subsection}{2}{\z@}%
                                     {-3.25ex\@plus -1ex \@minus -.2ex}%
                                     {1.5ex \@plus .2ex}%
                                     {\normalfont\Large\bfseries}}% from \large
\renewcommand\subsubsection{\@startsection{subsubsection}{3}{\z@}%
                                     {-3.25ex\@plus -1ex \@minus -.2ex}%
                                     {1.5ex \@plus .2ex}%
                                     {\normalfont\large\bfseries}}% from \normalsize
\makeatother
```

then save the `template.tex` file. Now knit your `index.Rmd` file to produce the PDF. 


I made a test chapter with this text:
```
# One pound symbol {#testing}

## Two pound symbols 

Some plain text here

### Three pound symbols 

Some plain text here

\subsection{LaTeX subsection}

Some plain text here

\subsubsection{LaTeX subsubsection}

Some plain text here

```
And got this output: 

![image](https://user-images.githubusercontent.com/1262179/39802422-d9504086-5322-11e8-922b-a608eeeb2555.png)

![image](https://user-images.githubusercontent.com/1262179/39802461-f74b0f62-5322-11e8-8f67-2b553a291aa6.png)

So that gives us graduated sizes for section headings, but we lose the small caps font. I tried a few things to get the small caps to change size, but they wouldn't budge. 

Let me know if that works for you. If it does, we'll merge it in to the template here so that future users can enjoy a bit more flexibility with their subheadings.

**Comment by amilesj** [comment]
Yes, that works for me and is much better. Thanks!

**Comment by amilesj** [comment]
One more question: I wanted to play with using sections to partition the document and tried un-commenting the specified lines in the `template.tex` file (starting with `\counterwithout`) but got the following error:

`! Undefined control sequence.
l.63 \setsecnumdepth
                    {subsubsection} `

Any suggestions?

**Comment by benmarwick** [comment]
Great to hear the sub-headings are now working as you expect. About the 'sections', I'm not 100% what you'd like them to look like, but I also found that un-commenting those lines didn't work. After a bit of trial and error I got to work with this:

```
%%%%%%%%%%%%%%%%%%
% If you want to use "sections" to partition your thesis
% un-comment the following:
%
\usepackage{chngcntr}
\counterwithout{section}{chapter}
\setcounter{secnumdepth}{2}
\def\sectionmark#1{\markboth{#1}{#1}}
\def\subsectionmark#1{\markboth{#1}{#1}}
\renewcommand{\thesection}{\arabic{section}}
\renewcommand{\thesubsection}{\thesection.\arabic{subsection}}
\makeatletter
\let\l@subsection\l@section
\let\l@section\l@chapter
\makeatother

\renewcommand{\thetable}{\arabic{table}}
\renewcommand{\thefigure}{\arabic{figure}}

%%%%%%%%%%%%%%%%%%
```

As far as I can see, the `\setsecnumdepth{subsubsection}` macro comes from the memoir class, but we are using the book class. And seems we need to the chngcntr package here also. 

But I'm not sure if the PDF output is what you're after, let me know.

**Comment by amilesj** [comment]
Thanks! I was just playing around with different looks...turns out that basically adds a running header for each section and changes the Table of Contents formatting a bit. I prefer it without, so I'll leave it commented. I did want to try out numbered sections and subsections, and was thinking maybe that is what the commented out code meant by "sections." 

I'm good with the changes you provided for progressively smaller subheadings for now. I'll let you know if I decide I want something else down the line and can't figure out how to implement it myself!

**Comment by amilesj** [comment]
Oh also for your merge to give more subsection flexibility in the PDFs: I ended up going with smallcaps for the section, regular for the subsection, and italics for the subsubsection, which I think gives a nicer visual structure than all normal font. I modified what you provided me to the following: 

```% Override the small caps headings so we have 
% section headers that decrease in size as we 
% use more # in the R Markdown
\makeatletter
\renewcommand\section{\@startsection {section}{1}{\z@}%
                                     {-3.5ex \@plus -1ex \@minus -.2ex}%
                                     {2.3ex \@plus.2ex}%
                                     {\normalfont\LARGE\scshape}}% from \Large
\renewcommand\subsection{\@startsection{subsection}{2}{\z@}%
                                     {-3.25ex\@plus -1ex \@minus -.2ex}%
                                     {1.5ex \@plus .2ex}%
                                     {\normalfont\Large\bfseries}}% from \large
\renewcommand\subsubsection{\@startsection{subsubsection}{3}{\z@}%
                                     {-3.25ex\@plus -1ex \@minus -.2ex}%
                                     {1.5ex \@plus .2ex}%
                                     {\normalfont\large\itshape}}% from \normalsize
\makeatother```

**Comment by PHUoL** [comment]
Is there a way to ensure that the formatting in TOC keeps the same structure. i.e. indentation of section, dotted lines connecting page numbers and reducing the spacing between section?

Thanks. Peter

**Comment by PHUoL** [comment]
I've also noticed that uncommenting for partitioning sections also results in number changes to, for example, the list of figures. Figure 3-1 is now shown as 1. The chapter number has been removed.

**Comment by benmarwick** [comment]
Hi Peter, I'm not sure, off the top of my head. I'll take a closer look. If you find anything in the meantime, do post an update here!

**Comment by PHUoL** [comment]
I have now sorted this out by adding \setcounter{tocdepth}{2} to line 766 of the uwthesis.cls.

Peter

**Comment by benmarwick** [comment]
@prcph thanks for the update, I'm impressed that you solved that so quickly!

**Comment by PHUoL** [comment]
I found the answer by googling and tried it out.

Peter

**Comment by Elesday** [comment]
I'd rather use this open issue since my question is about subsection!

I'd like to add the subsection (## in Rmd) number both in the text and in the ToC, how can I proceed? I already added the lines from your first answer here so that sections, sub, and subsubsections headers have different sizes.

Thanks a lot

PS: Unrelated question, but I'd like the figures to be in the middle of the text, not on their own page. Should I open a different issue to ask this question?
